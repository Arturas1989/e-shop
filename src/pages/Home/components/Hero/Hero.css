.hero {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(300px, 1fr));
  gap: var(--section-gap);
}

.hero > .column {
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: self-start;
  justify-content: center;
  gap: 20px;
}

.hero img {
  width: 100%;
  object-fit: cover;
}


