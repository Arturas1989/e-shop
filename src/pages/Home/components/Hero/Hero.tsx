import { PrimaryButton } from '../../../../components';
import Logo from './hero.jpg';

export const Hero = () => {
  return (
    <section className="hero">
      <div className="column">
        <h1>The cheapest products in the market</h1>
        <p>
          e-shop is the world's most popular and authoritative source for
          the highest variety of products. Find ratings and access to the newest products.
        </p>
        <PrimaryButton>
          <a href="/products">Explore products</a>
        </PrimaryButton>
      </div>
      <div className="column">
        <img src={Logo} alt="hero" />
      </div>
    </section>
  );
};
