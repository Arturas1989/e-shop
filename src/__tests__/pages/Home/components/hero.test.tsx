import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Hero } from '../../../../pages/Home/components/Hero/Hero';

describe('hero components renders', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Hero />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});