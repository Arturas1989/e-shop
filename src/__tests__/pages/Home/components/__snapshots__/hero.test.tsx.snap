// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hero components renders renders correctly 1`] = `
<section
  className="hero"
>
  <div
    className="column"
  >
    <h1>
      The cheapest products in the market
    </h1>
    <p>
      e-shop is the world's most popular and authoritative source for the highest variety of products. Find ratings and access to the newest products.
    </p>
    <button
      className="primary-button"
    >
      <a
        href="/products"
      >
        Explore products
      </a>
    </button>
  </div>
  <div
    className="column"
  >
    <img
      alt="hero"
      src="hero.jpg"
    />
  </div>
</section>
`;
